src/pages/benchviz/d3-util.ts

```diff
@@ -0,0 +1,27 @@
+import * as d3 from "d3";
+
+export function addLegend(
+    selection: d3.Selection<SVGSVGElement, unknown, null, undefined>,
+    items: Array<[string, string]>,
+) {
+    const legend = selection.append("g");
+
+    items.forEach(([name, color], i) => {
```

Comment:
IMO `for...of` is pretty much always preferred over `forEach`, since it aligns with regular loop code flow better

Reply:
Problem is I need the index (i) too, and keeping an extra counter variable seems uglier.

Reply:
You can get index with `.entries()`, we're using that a bunch of times in the main tstl codebase

```ts
for (const [index, [name, color]] of items.entries() {
```